src/main/java/com/avail/optimizer/jvm/JVMTranslator.java

```diff
@@ -1665,6 +1773,47 @@ void generateRunChunk ()
 			if (label != null)
 			{
 				method.visitLabel(label);
+				method.visitLineNumber(instruction.offset(), label);
+			}
+			if (callTraceL2AfterEveryInstruction)
+			{
+				loadReceiver(method);  // this, the executable chunk.
+				intConstant(method, instruction.offset());
+				// First line of the instruction toString
+				//noinspection DynamicRegexReplaceableByCompiledPattern
+				method.visitLdcInsn(instruction.toString().split("\\n", 2)[0]);
+				// Output the first read operand's value, as an Object, or null.
+				pushOneObject:
+				{
+					for (final L2Operand operand : instruction.operands())
+					{
+						if (operand instanceof L2ReadBoxedOperand)
+						{
+							load(
+								method,
+								((L2ReadBoxedOperand) operand).register());
+							break pushOneObject;
+						}
+						if (operand instanceof L2ReadIntOperand)
+						{
+							load(
+								method,
+								((L2ReadIntOperand) operand).register());
+							method.visitMethodInsn(
+								INVOKESTATIC,
+								getInternalName(Integer.class),
+								"valueOf",
+								getMethodDescriptor(
+									getType(Integer.class),
+									INT_TYPE),
+								false);
```

Comment:
You probably want to introduce a `CheckedMethod` for this.

Reply:
Done.  I intentionally only converted some of these, to see how well the mechanism worked.  It's a success, so I'll switch more of them to the new style, as I encounter them.